content/vendor/guides/persistent-datastores.md

```diff
@@ -0,0 +1,940 @@
+ 
+![View Update](/images/guides/kots/view-update.png)
+
+Once the update is deployed, we can head over to the Config tab and review our new toggle. You might also notice that we've removed the Ingress settings to simplify things for this guide:
+
+![Database Config](/images/guides/kots/database-config.png)
+
+Now that we have the configuration screen started, we can proceed to implement the "Embedded Postgres" option.
+
+* * *
+
+## Embedding a Database
+
+To implement the embedded Database option, we'll add a Kubernetes [Statefulset](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/), and use the [KOTS annotations for optional resources](/vendor/packaging/optional-resources/) to control when it will be included in the application.
+
+### Adding the Secret and StatefulSet
+
+Firse, we'll create a secret to store the root password for our embedded postgres instance. 
+
+```yaml
+# postgres-secret.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: postgres
+data:
+  DB_PASSWORD: '{{repl ConfigOption "embedded_postgres_password" | Base64Encode }}'
+```
+
+Next, create a new YAML file in your `manifests` directory with the following contents. Note the use of `kots.io/when` to only conditionally include this based on end-user inputs.
+
+```yaml
+# postgres-statefulset.yaml
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: postgres
+  labels:
+    app: pg-provider
+  annotations:
+    kots.io/when: '{{repl ConfigOptionEquals "postgres_type" "embedded_postgres" }}'
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: pg-provider
+  serviceName: postgres
+  template:
+    metadata:
+      labels:
+        app: pg-provider
+    spec:
+      containers:
+      - env:
+        - name: PGDATA
+          value: /var/lib/postgresql/data/pgdata
+        # create a db called "postgres"
+        - name: POSTGRES_DB
+          value: postgres
+        # create admin user with name "postgres"
+        - name: POSTGRES_USER
+          value: postgres
+        # use admin password from secret
+        - name: POSTGRES_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              key: DB_PASSWORD
+              name: postgres-secret
+        image: postgres:10
+        name: postgres
+        volumeMounts:
+        - mountPath: /var/lib/postgresql/data
+          name: pgdata
+      volumes:
+      - name: pgdata
+        persistentVolumeClaim:
+          claimName: pgdata
+  volumeClaimTemplates:
+  - metadata:
+      name: pgdata
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          storage: 1Gi
+```
+
+Finally, lets add a Service object so we can route traffic to our postgres instance, again using `kots.io/when` to conditionally include this resource.
+
+
+```yaml
+apiVersion: v1
```

Comment:
I think we should include the filename here, like is done in the other examples.